segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/ZooKeeperServiceRunner.java

```diff
@@ -106,19 +163,131 @@ public void stop() {
         } catch (Throwable e) {
             log.warn("Unable to cleanly shutdown ZooKeeper server", e);
         }
+
+        if (secureZK) {
+            System.clearProperty("zookeeper.serverCnxnFactory");
+            System.clearProperty("zookeeper.ssl.keyStore.location");
+            System.clearProperty("zookeeper.ssl.keyStore.password");
+            System.clearProperty("zookeeper.ssl.trustStore.location");
+            System.clearProperty("zookeeper.ssl.trustStore.password");
+        }
     }
 
     /**
      * Blocks the current thread and awaits ZooKeeper to start running locally on the given port.
      *
      * @param zkPort The ZooKeeper Port.
+     * @param secureZk Flag to notify whether the ZK is secure.
+     * @param trustStore Location of the trust store.
+     * @param keyStore Location of the key store.
+     * @param keyStorePasswordPath Location of password path for key store.
+     *                             Empty string if `secureZk` is false or a password does not exist.
+     * @param trustStorePasswordPath Location of password path for trust store.
+     *                               Empty string if `secureZk` is false or a password does not exist.
      * @return True if ZooKeeper started within a specified timeout, false otherwise.
      */
-    public static boolean waitForServerUp(int zkPort) {
-        val address = LOOPBACK_ADDRESS.getHostAddress() + ":" + zkPort;
-        return LocalBookKeeper.waitForServerUp(address, LocalBookKeeper.CONNECTION_TIMEOUT);
+    public static boolean waitForServerUp(int zkPort, boolean secureZk, String trustStore, String keyStore,
+                                          String keyStorePasswordPath, String trustStorePasswordPath) {
+        val address = "localhost:" + zkPort;
+        if (secureZk) {
+            return waitForSSLServerUp(address, LocalBookKeeper.CONNECTION_TIMEOUT, trustStore, keyStore,
+                    keyStorePasswordPath, trustStorePasswordPath);
+        } else {
+            return LocalBookKeeper.waitForServerUp(address, LocalBookKeeper.CONNECTION_TIMEOUT);
+        }
+    }
+
+    private static boolean waitForSSLServerUp(String address, long timeout, String trustStore, String keyStore,
+                                              String keyStorePasswdPath, String trustStorePasswordPath) {
+        //TODO: Create a ZK client to ensure that the server is up.
```

Comment:
Another `TODO` without an associated issue.

https://github.com/pravega/pravega/wiki/Contributing#todos

Reply:
This is already fixed. Removing `TODO`.